File: endpoints 23.48.57/photos.py
from silence.decorators import endpoint

@endpoint(
    route="/photos",
    method="GET",
    sql="SELECT * FROM Photos"
)
def get_all():
    pass

###############################################################################

@endpoint(
    route="/photos/index",
    method="GET",
    sql="SELECT * FROM Photos ORDER BY uploadDate DESC LIMIT 15"
)
def get_index():
    pass

###############################################################################

@endpoint(
    route="/photos/indexFollow/$userId",
    method="GET",
    sql="SELECT * FROM photos P JOIN usersfollow UF ON (P.userId = UF.user2) WHERE UF.user1 = $userId ORDER BY P.uploadDate DESC"
)
def get_index_follow():
    pass

###############################################################################

@endpoint(
    route="/photos/$photoId",
    method="GET",
    sql="SELECT * FROM Photos WHERE photoId = $photoId"
)
def get_by_id():
    pass

###############################################################################

@endpoint(
    route="/user/photos/$userId",
    method="GET",
    sql="SELECT * FROM Photos WHERE userId = $userId ORDER BY uploadDate DESC"
)
def get_by_UserId():
    pass

###############################################################################

@endpoint(
    route="/photos/category/$categoryId",
    method="GET",
    sql="SELECT * FROM photos NATURAL JOIN PhotoCategories WHERE categoryId = $categoryId GROUP BY photoId ORDER BY uploadDate DESC"
)
def get_by_Category():
    pass

###############################################################################

@endpoint(
    route="/photos",
    method="POST",
    sql="INSERT INTO Photos (title, description, photoURL, visibility, userId) VALUES ($title, $description, $photoURL, $visibility, $userId)",
    description="Creates a new photo",
    auth_required=True,
)
def create(title, description, photoURL, visibility, userId):
    pass

###############################################################################

@endpoint(
    route="/photos/$photoId",
    method="PUT",
    sql="UPDATE Photos SET title = $title, description = $description, photoURL = $photoURL, visibility = $visibility WHERE photoId = $photoId",
    description="Updates an existing photo",
    auth_required=True,
)
def update(title, description, photoURL, visibility):
    pass

###############################################################################

@endpoint(
    route="/photos/$photoId",
    method="DELETE",
    sql="DELETE FROM Photos WHERE photoId = $photoId",
    description="Removes a photo",
    auth_required=True,
)
def delete():
    pass

File: docs/js/api/photos.js
"use strict";

import { BASE_URL, requestOptions } from "./common.js";

const photosAPI = { 
    getAll: function () {
        return new Promise(function (resolve , reject) { 
            axios
                .get(`${BASE_URL}/photos/index`, requestOptions) 
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        }); 
    },

    getAllMyFollows: function (userId) {
        return new Promise(function (resolve , reject) { 
            axios
                .get(`${BASE_URL}/photos/indexFollow/${userId}`, requestOptions) 
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        }); 
    },

    getById: function (photoId) {
        return new Promise(function (resolve , reject) {
            axios
                .get(`${BASE_URL}/photos/${photoId}`, requestOptions) 
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        }); 
    },

    getByUser: function (userId) {
        return new Promise(function (resolve , reject) {
            axios
                .get(`${BASE_URL}/user/photos/${userId}`, requestOptions) 
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        }); 
    },

    create: function (formData) { 
        return new Promise(function (resolve , reject) {
            axios
                .post(`${BASE_URL}/photos`, formData , requestOptions) 
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        }); 
    },
    
    update: function (photoId, formData) {
        return new Promise(function (resolve , reject) {
            axios
                .put(`${BASE_URL}/photos/${photoId}`, formData ,requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        }); 
    },
        
    delete: function (photoId) {
        return new Promise(function (resolve , reject) {
            axios
                .delete(`${BASE_URL}/photos/${photoId}`, requestOptions)
                .then(response => resolve(response.data))
                .catch(error => reject(error.response.data.message));
        }); 
    },
};

export { photosAPI };